pkg/reconciler/v1alpha1/stack/stack.go

```diff
@@ -123,17 +126,53 @@ func (c *Reconciler) getStackImages(stackSpec expv1alpha1.StackSpec) (string, st
 		return "", "", err
 	}
 
-	_, baseImageId, err := c.ImageFetcher.Fetch(keychain, stackSpec.BuildImage.Image)
+	buildImage, buildImageId, err := c.ImageFetcher.Fetch(keychain, stackSpec.BuildImage.Image)
 	if err != nil {
 		return "", "", err
 	}
 
-	_, runImageId, err := c.ImageFetcher.Fetch(keychain, stackSpec.RunImage.Image)
+	runImage, runImageId, err := c.ImageFetcher.Fetch(keychain, stackSpec.RunImage.Image)
 	if err != nil {
 		return "", "", err
 	}
 
-	return baseImageId, runImageId, nil
+	err = validateStackId(stackSpec.Id, buildImage, runImage)
+	if err != nil {
+		return "", "", err
+	}
+
+	return buildImageId, runImageId, nil
+}
+
+func validateStackId(stackId string, buildImage ggcrv1.Image, runImage ggcrv1.Image) error {
+	buildStack, err := getStackLabel(buildImage)
+	if err != nil {
+		return err
+	}
+
+	runStack, err := getStackLabel(runImage)
+	if err != nil {
+		return err
+	}
+
+	if (buildStack == stackId) && (runStack == stackId) {
+		return nil
+	} else {
+		return errors.New(fmt.Sprintf("Invalid stack images. Expected stack: %s, Build Image stack: %s, Run Image stack: %s", stackId, buildStack, runStack))
```

Comment:
Could you use `errors.Errorf` instead of `errors.New(fmt.Sprintf` ?

Reply:
Also, super minor but, could we avoid capitalizing the error string? 
https://github.com/golang/go/wiki/CodeReviewComments#error-strings